### PR TITLE
update boundary mode in profile_line

### PR DESCRIPTION
**Pull request by grlee77**
## Description

This PR resolves a TODO item related to switching the default `profile_line` boundary mode to 'reflect' for release 0.19. I also added a small change to fix the behavior for modes 'constant' and 'wrap' using their new implementations from SciPy 1.6 when available. This fixes, for example, the unexpected behavior raised at the top of gh-4279.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Review by jni — APPROVED** [review]
❤️

File: skimage/_shared/utils.py
Comment (sciunto) [inline skimage/_shared/utils.py:427]: As for the other PR, I think a bullet in the todo list for next year would be helpful. :)

Comment (grlee77) [reply skimage/_shared/utils.py:427]: Sure. I just reorganized that TODO section in #5173, though, so I think I will wait until we merge that one and then rebase here and add it

**Review by rfezzani — APPROVED** [review]
Thank you @grlee77, it seems that we need a rebase before merging :wink:

File: skimage/_shared/utils.py
Comment (grlee77) [reply skimage/_shared/utils.py]: I rebased now that #5173 was merged and a TODO has been added

**Comment by rfezzani** [comment]
Thank you @grlee77 :tada:

**Pull request merged**
